forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -619,7 +619,7 @@ private Card changeZone(final Zone zoneFrom, Zone zoneTo, final Card c, Integer
         checkStaticAbilities();
 
         // 400.7g try adding keyword back into card if it doesn't already have it
-        if (zoneTo.is(ZoneType.Stack) && cause != null && cause.isSpell() && c.equals(cause.getHostCard())) {
+        if (zoneTo.is(ZoneType.Stack) && cause != null && cause.isSpell() && !cause.isIntrinsic() && c.equals(cause.getHostCard())) {
```

Comment:
Sorry, still found a way to break it:

When it gains two flashback (e.g. Flicker Snapcaster after resolving) and then cast it via the _first_ altcost it won't be exiled

Works with any spell

Reply:
you don't need to flicker it, just cast a second Snapcaster

i just don't understand it why it gets two times the keywords when moving to stack, i need to debug it more

Reply:
@tool4ever found the reason: getReplacementEffects returns an FCollection meaning it does filter duplicates

meaning Line 270+ with `copied.setChangedCardKeywords(c.getChangedCardKeywords());` already added the Flashkeyword to stack, but as a copy not as direct value (keywords doesn't have ID yet)
i made it so it doesn't copy when moving to stack


Can we try to get back why this was done?
a624cf565a6adf3fd12d968311acecd99cbdd4c9

Reply:
From the timing I'd say we had the same issue then :P

![image](https://user-images.githubusercontent.com/8506892/218314332-da0ea58d-0fe8-4b62-97ff-c4f6a8021825.png)

Reply:
the Old Code for Flashback isn't needed anymore now that the new code especially can handle the CastKeyword